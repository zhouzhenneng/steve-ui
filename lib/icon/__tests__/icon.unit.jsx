import 'jsdom-global/register';
import * as renderer from 'react-test-renderer';
import * as React from 'react'
import Icon from '../icon';
import {mount} from 'enzyme';

describe('icon', () => {
  it('render successfully', ()=>{
    const json = renderer.create(<Icon  name="twitter"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('onclick', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="twitter" onClick={fn}/>);
    component.find('svg').simulate("click")
    expect(fn).toBeCalled();
  })
})