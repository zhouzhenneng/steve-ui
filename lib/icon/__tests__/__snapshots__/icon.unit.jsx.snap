// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<svg
  className="steve-icon"
>
  <use
    xlinkHref="#twitter"
  />
</svg>
`;
